### DISCUSSION TITLE
Why doesn't useMutation have a select option? This would reduce the amount of related formatting code.

### DISCUSSION THREAD

**Discussion by Deer404** [TanStack/query#6396] [Q&A]
Why doesn't useMutation have a select option? This would reduce the amount of related formatting code.

**Comment by Deer404** [comment]
Is there a problem with my needs?

**Comment by TkDodo** [comment]
Mutations work differently from queries. You can't really have multiple observers on the same mutations, so transformations with select (which are a render optimization mostly) aren't necessary.
